pkg/manager/internal/controllers/project_controller.go

```diff
@@ -113,39 +110,24 @@ func (r *ProjectReconciler) reconcileMembers(ctx context.Context, project *stora
 	return nil
 }
 
-// Project Namespace name is organization_name-separator-project_name
-func generateProjectNamespaceName(organizatioName, projectName string) string {
-	return fmt.Sprintf("%s%s%s", organizatioName, projectNamespaceNameSeparator, projectName)
-}
-
-// Project Namespace name is organization_name-separator-project_name
-func deconstructProjectNamespaceName(projectNamespaceName string) (string, string, error) {
-	split := strings.Split(projectNamespaceName, projectNamespaceNameSeparator)
-	if len(split) != 2 {
-		return "", "", fmt.Errorf("not a valid project namespace name to deconstruct: %s", projectNamespaceName)
-	}
-
-	return split[0], split[1], nil
-}
-
-func (r *ProjectReconciler) SetupWithManager(mgr ctrl.Manager) error {
-	enqueuer := owner.EnqueueRequestForOwner(&storagev1alpha1.Project{}, mgr.GetScheme())
+func (r *ProjectReconciler) SetupWithManager(mgr ctrl.Manager, log logr.Logger) error {
+	enqueuer := owner.EnqueueRequestForOwner(&storagev1alpha1.Project{}, mgr.GetScheme()).(*handler.EnqueueRequestsFromMapFunc)
 
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&storagev1alpha1.Project{}).
 		Watches(&source.Kind{Type: &corev1.Namespace{}}, enqueuer).
 		Watches(&source.Kind{Type: &rbacv1.RoleBinding{}}, &handler.EnqueueRequestsFromMapFunc{
 			ToRequests: handler.ToRequestsFunc(func(object handler.MapObject) []reconcile.Request {
-				org, pro, err := deconstructProjectNamespaceName(object.Meta.GetNamespace())
-				if err != nil {
+				ctx := context.Background()
+				ns := &corev1.Namespace{}
+				if err := r.Client.Get(ctx, types.NamespacedName{Name: object.Meta.GetNamespace()}, ns); err != nil {
+					log.Error(err, fmt.Sprintf("failed to get rolebinding namespace %q", object.Meta.GetNamespace()))
```

Comment:
rather panic on this error since it's pretty severe if you cannot get ns. the pod shall restart and all is good.

Reply:
done